UPGRADE_NOTES.md

```diff
@@ -1,4 +1,250 @@
+    settings.logger = logger
+  end
+  ```
+</details>
+- Update: `script/setup` to include the new postgres checks.
+```diff
++ printf "\n▸ Checking that postgres is installed\n"
++ check_postgres | indent
++ printf "✔ Done\n" | indent
+
++ printf "\n▸ Verifying postgres connection\n"
++ lucky db.verify_connection | indent
+
+printf "\n▸ Setting up the database\n"
+lucky db.create | indent
+```
+
+### Database updates
+- Add: a new `AppDatabase` class in `src/app_database.cr` that inherits from `Avram::Database`.
+```crystal
+class AppDatabase < Avram::Database
+end
+```
+- Add: `require "./app_database"` to `src/app.cr` right below the `require "./shards"`.
+- Rename: `Avram::Repo.configure` to `AppDatabase.configure` in `config/database.cr`.
+- Add: `Avram.configure` block.
+<details>
+  <summary>config/database.cr</summary>
+
+  ```crystal
+  database_name = "..."
+
+  AppDatabase.configure do |settings|
+    if Lucky::Env.production?
+      settings.url = ENV.fetch("DATABASE_URL")
+    else
+      settings.url = ENV["DATABASE_URL"]? || Avram::PostgresURL.build(
+        database: database_name,
+        hostname: ENV["DB_HOST"]? || "localhost",
+        # Some common usernames are "postgres", "root", or your system username (run 'whoami')
+        username: ENV["DB_USERNAME"]? || "postgres",
+        # Some Postgres installations require no password. Use "" if that is the case.
+        password: ENV["DB_PASSWORD"]? || "postgres"
+      )
+    end
+  end
+
+  Avram.configure do |settings|
+    settings.database_to_migrate = AppDatabase
+
+    # this is moved from your old `Avram::Repo.configure` block.
+    settings.lazy_load_enabled = Lucky::Env.production?
+  end
+  ```
+</details>
+
+- Move: the `settings.lazy_load_enabled` from `AppDatabase.configure` to `Avram.configure` block.
+- Add: a `database` class method to `src/models/base_model.cr` that returns `AppDatabase`.
+```crystal
+abstract class BaseModel < Avram::Model
+  def self.database : Avram::Database.class
+    AppDatabase
+  end
+end
+```
+- Update: `Avram::Repo` to `AppDatabase` in `spec/setup/clean_database.cr`.
+- Update: Any model that uses `UUID` for a primary key must use the new `primary_key` syntax.
+```crystal
+class User < BaseModel
+  # Before migration
+  table :users, primary_key_type: :uuid do
+    column email : String
+  end
+
+  # After migration
+  table :users do
+    primary_key id : UUID
+  end
+end
+```
+- Update: models now have a `default_columns` macro that adds `primary_key id : Int64` and `timestamps`. This macro must be overriden if your tables use a different column type for your primary keys.
+```crystal
+abstract class BaseModel < Avram::Model
+  macro default_columns
+    primary_key id : Int32
+    timestamps
+  end
+end
+```
+- Note: Avram now defaults primary keys to `Int64` instead of `Int32`. You can use the `change_type` macro to migrate your primary keys to `Int64` if you need. Run `lucky gen.migration UpdatePrimaryKeyTypes`.
```

Comment:
I think we should move this above and make it clear this is the recommended path. It’ll make everything easier going forward if they convert now